R/SoilProfileCollection-methods.R

```diff
@@ -411,23 +414,72 @@ setMethod("subsetHz", signature(x = "SoilProfileCollection"), function(x, ...) {
   subcrit <- Reduce('&', res)
   
   if (!is.logical(subcrit)) {
-    badxpr <- paste0("'",paste0(.dots[sapply(.dots, function(x) !is.logical(x))],
-                                collapse=",'"),"'")
+    badxpr <- paste0("'", paste0(.dots[sapply(.dots, function(x) {
+      !is.logical(x)
+    })], collapse = ",'"), "'")
     message(sprintf("%s is not logical; returning `x` unchanged", badxpr))
     return(x)
   }
   
   newhz <- .data[which(subcrit),]
   
-  # subset SPC first to remove sites and other slots
-  x <- x[which(profile_id(x) %in% newhz[[idname(x)]]),]
+  if (drop) {
+    # subset SPC first to remove sites and other slots
+    x <- x[which(pid %in% newhz[[idn]]), ]
+    
+    # then replace horizons with horizon subset 
+    replaceHorizons(x) <- newhz
+  } else {
+    # reinsert empty horizons and site data for "retained" profiles
+    x <- .insert_dropped_horizons(x, newhz)
```

Comment:
OK, just to be sure: this replaces `x` entirely. This requires special attention to ensure the resulting SPC isn't altered in unexpected ways. Would it be simpler / safer to implement via `replaceHorizons()` on a modified `newhz`?

Reply:
There are two modes of usage of the internal function... one that returns a SPC (replaceHorizons() is used internally) and another that returns just a list of the data.frames from site and horizon.